server/src/main/java/org/apache/iotdb/db/query/dataset/DeviceIterateDataSet.java

```diff
@@ -80,39 +84,39 @@
   private int[] currentColumnMapRelation;
   private Map<Path, TSDataType> tsDataTypeMap;
 
-  public DeviceIterateDataSet(QueryPlan queryPlan, QueryContext context,
+  public DeviceIterateDataSet(AlignByDevicePlan alignByDevicePlan, QueryContext context,
       IQueryRouter queryRouter) {
-    super(null, queryPlan.getDataTypes());
+    super(null, alignByDevicePlan.getDataTypes());
 
     // get deduplicated measurement columns (already deduplicated in TSServiceImpl.executeDataQuery)
-    this.deduplicatedMeasurementColumns = queryPlan.getMeasurements();
-    this.tsDataTypeMap = queryPlan.getDataTypeMapping();
+    this.deduplicatedMeasurementColumns = alignByDevicePlan.getMeasurements();
+    this.tsDataTypeMap = alignByDevicePlan.getDataTypeMapping();
     this.queryRouter = queryRouter;
     this.context = context;
-    this.measurementColumnsGroupByDevice = queryPlan.getMeasurementsGroupByDevice();
-    this.deviceToFilterMap = queryPlan.getDeviceToFilterMap();
-    this.notExistMeasurements = queryPlan.getNotExistMeasurements();
-    this.constMeasurements = queryPlan.getConstMeasurements();
-    this.positionOfNotExistMeasurements = queryPlan.getPositionOfNotExistMeasurements();
-    this.positionOfConstMeasurements = queryPlan.getPositionOfConstMeasurements();
+    this.measurementColumnsGroupByDevice = alignByDevicePlan.getDeviceToMeasurementsMap();
+    this.deviceToFilterMap = alignByDevicePlan.getDeviceToFilterMap();
+    this.notExistMeasurements = alignByDevicePlan.getNotExistMeasurements();
+    this.constMeasurements = alignByDevicePlan.getConstMeasurements();
+    this.positionOfNotExistMeasurements = alignByDevicePlan.getPositionOfNotExistMeasurements();
+    this.positionOfConstMeasurements = alignByDevicePlan.getPositionOfConstMeasurements();
     //BuildOutDataTypes();
 
-    if (queryPlan instanceof GroupByPlan) {
+    if (alignByDevicePlan.getGroupByPlan() != null) {
       this.dataSetType = DataSetType.GROUPBY;
       // assign parameters
-      this.unit = ((GroupByPlan) queryPlan).getInterval();
-      this.slidingStep = ((GroupByPlan) queryPlan).getSlidingStep();
-      this.startTime = ((GroupByPlan) queryPlan).getStartTime();
-      this.endTime = ((GroupByPlan) queryPlan).getEndTime();
+      this.unit = alignByDevicePlan.getGroupByPlan().getInterval();
```

Comment:
Reuse the query plan in AlignByDevicePlan, no need to set the parameters again and again

Reply:
Good Idea. Fixed.